doc/specs/vulkan/chapters/fundamentals.txt

```diff
@@ -286,8 +286,8 @@ objects, and not to destroy them while they are still in use.
 
 Application-owned memory is immediately consumed by any Vulkan command it is
 passed into.
-The application can: alter or free this memory as soon as the commands that
-consume it have returned.
+The application must: not write to nor free this memory before the command
+that consumed it have returned.
```

Comment:
Well, I will try to

> you can't really do in any C API

I wish. You **can** though. And if some concurrent situations it does even make sense.
It would made some sense if a `VkFence` was signalled, and there would be a doubt whether the pointer is still needed by the command even if it hasn't already returned.
For that matter you can even write `const` in C API.